src/dmd/trace.d

```diff
@@ -0,0 +1,71 @@
+/**
+ * Compiler implementation of the
+ * $(LINK2 http://www.dlang.org, D programming language).
+ *
+ * Copyright:   Copyright (C) 1999-2018 by The D Language Foundation, All Rights Reserved
+ * Authors:     Stefan Koch
+ * License:     $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/trace.d, _trace.d)
+ * Documentation:  https://dlang.org/phobos/dmd_trace.html
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/src/dmd/trace.d
+ */
+
+module dmd.trace;
+import dmd.dsymbol;
+
+enum SYMBOL_TRACE = true;
+
+struct SymbolProfileEntry
+{
+    Dsymbol sym;
+    ulong begin_ticks;
+    ulong end_ticks;
+
+    ulong begin_mem;
+    ulong end_mem;
+
+    string kind;
+    string fn;
+}
+
+extern (C) __gshared uint dsymbol_profile_array_count;
+extern (C) __gshared SymbolProfileEntry* dsymbol_profile_array;
+enum dsymbol_profile_array_size = ushort.max * 1024; // 64 million entries should do, no ?
```

Comment:
I had some trouble with memory consumption. When i changed this to
```enum dsymbol_profile_array_size = ushort.max * 512;```
I was still able to compile everytning and run the unittests. You might want to think about dynamically resizing on demand, instead of hardcoding this.